python/plugins/db_manager/db_plugins/postgis/connector.py

```diff
@@ -348,6 +348,13 @@ def getInfo(self):
         self._close_cursor(c)
         return res
 
+    def getPsqlVersion(self):
+        regex = r"^PostgreSQL\s([0-9]{1,2})"
+        match = re.match(regex, self.getInfo()[0])
+        if match:
+            return int(match.group(1))
+        return
```

Comment:
Should this give a meaningful error if no match?

Reply:
Ideally, it should.
But I looked at other functions and none of them raise an exception.

https://github.com/qgis/QGIS/blob/5d58e8ac917cf8201faba63686cc81cd9c6fc300/python/plugins/db_manager/db_plugins/postgis/connector.py#L351-L369

I replaced latest `return` with `raise DbError("Unrecognized PostgreSQL version")` and this error is raised quite late when using DB Manager (only when showing layer info and not at connection).
![image](https://github.com/qgis/QGIS/assets/39594821/dd448e2f-3330-4d1e-b0ab-7a7852ef1633)

I can commit this change if you want but all other functions should be updated and it's not in my plan to do it.

Reply:
I think that's better than showing it nowhere?
I would add the unparsable string to the error message to make it easier fixable if raised.

Reply:
Done in https://github.com/qgis/QGIS/pull/54401/commits/8fb3352348d810aea73f16e77d807acd2f8a8b3f